cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/config/BrokerSetFileResolver.java

```diff
@@ -44,7 +44,7 @@ public class BrokerSetFileResolver implements BrokerSetResolver {
 
   @Override
   public void configure(Map<String, ?> configs) {
-    KafkaCruiseControlConfig config = new KafkaCruiseControlConfig(configs);
+    KafkaCruiseControlConfig config = new KafkaCruiseControlConfig(configs, false);
```

Comment:
Why do we even create a new `KafkaCruiseControlConfig` from the passed parameter. Can we avoid this?

Reply:
The passed parameter is a Map. Several lines below it used the KafkaCruiseControlConfig object to create a instance:


`        config.getConfiguredInstance(AnalyzerConfig.BROKER_SET_ASSIGNMENT_POLICY_CLASS_CONFIG, BrokerSetAssignmentPolicy.class);`

So I think it's reasonable to use KafkaCruiseControlConfig here.

Reply:
@CCisGG is correct. This was done to load the instance and reuse the ``getConfiguredInstance()`` method. I was trying to emulate ``BrokerCapacityConfigFileResolver`` to keep up with the repo standards. 

However, ``BrokerCapacityConfigFileResolver`` just loaded the json file name from `` _configFile = KafkaCruiseControlUtils.getRequiredConfig(configs, CAPACITY_CONFIG_FILE);``.

``KafkaCruiseControlUtils`` did NOT have a method to get the instance. Hence I created an instance of ``KafkaCruiseControlConfig`` to reuse the ``getConfiguredInstance`` method.

Reply:
Creating a new `KafkaCruiseControlConfig` is expensive and I suspect we can avoid it.
 
-- i.e. we can create the `BrokerSetAssignmentPolicy` instance before creating this class in. For examples of this usage, please search for the use of `parameterConfigOverrides` with `*_OBJECT_CONFIG;` for `getConfiguredInstance`.

Reply:
@efeg I did a code search for parameterConfigOverrides and getConfiguredInstance, but failed to find an valid example for our case. All the examples are calling the getConfiguredInstance method on an KafkaCruiseControlConfig object. Would you mind provide a concrete example?

Reply:
They are supposed to use KafkaCruiseControlConfig object. The `BrokerSetAssignmentPolicy` object should be created from within `BalancingConstraint` constructor using the parameter of the constructor,  which is a `KafkaCruiseControlConfig`. Then you should update `_brokerSetResolver = config.getConfiguredInstance(AnalyzerConfig.BROKER_SET_RESOLVER_CLASS_CONFIG, BrokerSetResolver.class);` to get the extra parameter (i.e. `Map<String, Object> configOverrides`).

Reply:
Addressed